pkg/sensors/exec/cgroups_test.go

```diff
@@ -70,6 +117,125 @@ func cgroupRmdir(t *testing.T, cgroupfsPath string, hierarchy string, dir string
+		"cgroup.IDTracker": msg.CgrpidTracker,
+		"cgroup.ID":        msg.Cgrpid,
+		"cgroup.state":     st,
+		"cgroup.level":     msg.CgrpData.Level,
+		"cgroup.path":      cgrpPath,
+	}).Info("Received Cgroup event")
+
+	// match only our target cgroup paths
+	if strings.HasPrefix(cgrpPath, filepath.Join("/", cgroupHierarchy[0].path)) == false {
+		return ops.CgroupOpCode(ops.MSG_OP_CGROUP_UNDEF), ""
+	}
+
+	require.NotZero(t, msg.PID)
+	require.NotZero(t, msg.Cgrpid)
+	require.NotZero(t, msg.CgrpidTracker)
+	require.NotEqualValues(t, msg.CgrpData.State, ops.CGROUP_UNTRACKED)
+	require.NotZero(t, msg.CgrpData.Level)
+
+	return op, cgrpPath
+}
+
+func requireCgroupEventOpMkdir(t *testing.T, msg *grpcexec.MsgCgroupEventUnix, cgrpMapPath string) {
+	require.EqualValues(t, ops.CGROUP_NEW, msg.CgrpData.State)
+
+	looked, err := cgrouptrackmap.LookupTrackingCgroup(cgrpMapPath, msg.CgrpidTracker)
+	assert.NoError(t, err)
+	if looked == nil {
+		t.Fatalf("Failed to find tracking cgroupID=%d in bpf-map=%s", msg.CgrpidTracker, cgrpMapPath)
+	}
+	require.EqualValues(t, msg.CgrpData.Level, looked.Level)
+	require.EqualValues(t, msg.CgrpData.Name, looked.Name)
+}
+
+func requireCgroupEventOpRmdir(t *testing.T, msg *grpcexec.MsgCgroupEventUnix, cgrpMapPath string) {
+	looked, err := cgrouptrackmap.LookupTrackingCgroup(cgrpMapPath, msg.CgrpidTracker)
+	assert.Error(t, err)
+	if looked != nil {
+		t.Fatalf("Failed found tracking cgroup with cgroupID=%d in bpf-map=%s", msg.CgrpidTracker, cgrpMapPath)
+	}
+}
+
+func assertCgroupEvents(ctx context.Context, t *testing.T, cgroupMount string, cgroupHierarchy []cgroupHierarchy, trackedLevel uint32, trigger func()) {
+	cgrpMapPath := filepath.Join(bpf.MapPrefixPath(), testsensor.GetCgroupsTrackingMap().Name)
+	events := perfring.RunTestEvents(t, ctx, trigger)
+	for _, ev := range events {
+		if msg, ok := ev.(*grpcexec.MsgCgroupEventUnix); ok {
+			if msg.Common.Op == ops.MSG_OP_CGROUP {
+				op, cgrpPath := getCgroupEventOpAndPath(t, msg, cgroupHierarchy)
+				// match only our target cgroup paths
+				if cgrpPath == "" {
+					continue
+				}
+				cgrpName := filepath.Base(cgrpPath)
+				require.Equal(t, cgrpName, cgroups.CgroupNameFromCStr(msg.CgrpData.Name[:processapi.CGROUP_NAME_LENGTH]))
+
+				if msg.CgrpData.Level > trackedLevel {
+					t.Fatalf("Error received cgroup event below tracked cgroup hierarchy, level:%d , path:%s",
+						msg.CgrpData.Level, cgrpPath)
+				}
+				switch op {
+				case ops.MSG_OP_CGROUP_MKDIR:
+					require.EqualValues(t, ops.CGROUP_NEW, msg.CgrpData.State)
+					require.EqualValues(t, cgroupHierarchy[msg.CgrpData.Level-1].path, cgrpName)
+
+					requireCgroupEventOpMkdir(t, msg, cgrpMapPath)
+					cgroupHierarchy[msg.CgrpData.Level-1].added = true
+				case ops.MSG_OP_CGROUP_RMDIR:
+					require.EqualValues(t, cgroupHierarchy[msg.CgrpData.Level-1].path, cgrpName)
+					requireCgroupEventOpRmdir(t, msg, cgrpMapPath)
+					cgroupHierarchy[msg.CgrpData.Level-1].removed = true
+				}
```

Comment:
Yes, here we set the `cgroupHierarchy[msg.CgrpData.Level-1].{added|removed} = true`  , so later
check function `assertCgroupDirTracking()` that asserts if the related directories were tracked added and removed are set to true, otherwise fail.

Reply:
Or check the end of the Test function where we iterate over defaultKubeCgroupHierarchy to assert the events about cgroup directories being created and deleted, following patch turn this logic into a specific shared function `assertCgroupDirTracking()`